ModsenOnlineStore.Store.Infrastructure/Data/ProductRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ModsenOnlineStore.Store.Application.Interfaces.ProductInterfaces;
+using ModsenOnlineStore.Store.Domain.Entities;
+
+namespace ModsenOnlineStore.Store.Infrastructure.Data
+{
+    public class ProductRepository : IProductRepository
+    {
+        private readonly DataContext context;
+
+        public ProductRepository(DataContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<List<Product>> GetAllProducts()
+        {
+            return await context.Products.AsNoTracking().ToListAsync();
+        }
+
+        public async Task<Product?> GetProductById(int id)
+        {
+            return await context.Products.AsNoTracking().FirstOrDefaultAsync(p => p.Id == id);
+        }
+
+        public async Task<List<Product>> AddProduct(Product product)
+        {
+            context.Products.Add(product);
+            await context.SaveChangesAsync();
+            return await GetAllProducts();
+        }
+
+        public async Task<List<Product>> UpdateProduct(Product product)
```

Comment:
why do you return all products after update

Reply:
fixed